plugins/mimecast/komand_mimecast/util/api.py

```diff
@@ -0,0 +1,186 @@
+import base64
+import datetime
+import hashlib
+import hmac
+import json
+import uuid
+
+import requests
+
+from insightconnect_plugin_runtime.exceptions import PluginException
+from komand_mimecast.util.util import Utils
+from komand_mimecast.util.constants import (
+    API,
+    DATA_FIELD,
+    META_FIELD,
+    XDK_BINDING_EXPIRED_ERROR,
+    CODE,
+    FIELD_VALIDATION_ERROR,
+    VALIDATION_BLANK_ERROR,
+    ERROR_CASES,
+    BASIC_ASSISTANCE_MESSAGE,
+    FAIL_FIELD,
+    STATUS_FIELD,
+)
+
+
+class MimecastAPI:
+    def __init__(self, region: str, access_key: str, secret_key: str, app_id: str, app_key: str, logger=None):
+        self.url = Utils.prepare_base_url(region)
+        self.access_key = access_key
+        self.secret_key = secret_key
+        self.app_id = app_id
+        self.app_key = app_key
+        self.logger = logger
+        self.logger.info(self.url)
+
+    def get_managed_url(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/ttp/url/get-all-managed-urls", data=data)
+
+    def get_ttp_url_logs(self, data: dict, meta_data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/ttp/url/get-logs", data=data, meta_data=meta_data)
+
+    def add_group_member(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/directory/add-group-member", data=data)
+
+    def create_blocked_sender_policy(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/policy/blockedsenders/create-policy", data=data)
+
+    def create_managed_url(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/ttp/url/create-managed-url", data=data)
+
+    def decode_url(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/ttp/url/decode-url", data=data)
+
+    def delete_group_member(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/directory/remove-group-member", data=data)
+
+    def delete_managed_url(self, data: dict) -> dict:
+        return self._handle_rest_call("POST", f"{API}/ttp/url/delete-managed-url", data)
```

Comment:
This is inconsequential, but is there any particular reason to pass `data` to an unnamed input here, unlike most of the other methods where `data=data`?

Reply:
There is no reason for this. I will fix that and we will have the same format everywhere :)